### ISSUE TITLE
Bower.json of dep package is not retrieved at the right moment

### ISSUE THREAD

**Issue by WouterJ** [Bee-Lab/bowerphp#115]
The bower.json of dep packages is retrieved pretty early in the process, even before the version is resolved. This means that 99% of the time, [`Repository#getBower()`](https://github.com/Bee-Lab/bowerphp/blob/master/src/Bowerphp/Repository/GithubRepository.php#L58) is called with a version constraint instead of an exact version (e.g. `1.7.*`). It then tries to request `/repos/jqyery/jquery/contents/bower.json?ref=1.7.*`, which obviously fails. It then [fallsback to get `bower.json` from master](https://github.com/Bee-Lab/bowerphp/blob/master/src/Bowerphp/Repository/GithubRepository.php#L185).

This is a problem with jQuery, as it only includes `bower.json` files in their releases and not in their running branches.

It's even a bigger problem when the `bower.json` in master differs from the one in a release. As it's requesting it with a version constraint, it'll fail and the master one would be used. This could result in completely wrong dependencies.

In conclusion:

* First resolve the version and then try to get the bower.json (even better: get the bower.json from the downloaded zip file, this saves a slow request)
* Do not fallback to master if it didn't exists in the specified version

**Comment by anroots** [comment]
Would this be the reason why I get the "source" version of jQuery that does not include a "dist" folder when installing with bowerphp?

**Comment by WouterJ** [comment]
Yes

**Comment by garak** [comment]
No :disappointed: See #114

**Comment by WouterJ** [comment]
Ah, indeed. Sorry, mixed up the issues

**Comment by garak** [comment]
Ehi @WouterJ can you check the last commit? Now we should get the version *before* getting the ``bower.json`` file, so jquery should be fine now.
Indeed, I'm not sure of understanding why we shouldn't get ``bower.json`` from master as fallback: in this case, it's the only info we could get, and it can't be different from version's one (since the latter doesn't exist). Feel free to tell me more to explain. Thanks.

**Comment by WouterJ** [comment]
It now works as expected.
